packages/frontend/hooks/scaffold-eth/useEthPrice.ts

```diff
@@ -0,0 +1,35 @@
+import { Fetcher, Route, Token, WETH } from "@uniswap/sdk";
+import { useEffect, useState } from "react";
+import { useProvider } from "wagmi";
+
+// Get the price of ETH based on ETH/DAI trading pair from Uniswap SDK
+export const useEthPrice = () => {
+  const provider = useProvider({ chainId: 1 });
+  const [ethPrice, setEthPrice] = useState<number>(0);
+
+  useEffect(() => {
+    Promise.resolve(
+      fetchPriceFromUniswap(provider).then((price: any) => {
+        console.log("price", price);
+        setEthPrice(price);
+      }),
+    );
+  }, [provider]);
```

Comment:
We could do everything in the useEffect.

- No need to pass the provider as an argument, we already have it in the state.

```ts
  useEffect(() => {
    const fetchPriceFromUniswap = async () => {
      try {
        const DAI = new Token(1, "0x6B175474E89094C44Da98b954EedeAC495271d0F", 18);
        const pair = await Fetcher.fetchPairData(DAI, WETH[DAI.chainId], provider);
        const route = new Route([pair], WETH[DAI.chainId]);
        const price = parseFloat(route.midPrice.toSignificant(6));

        setEthPrice(price);
      } catch (error) {
        console.log("Error fetching ETH price from Uniswap: ", error);
      }
    };
    fetchPriceFromUniswap(); // no need to await here.
  }, [provider]);
```

Reply:
Unresolving this. I don't see this changes implemented. Maybe you forgot to push?

Reply:
this is not what i see

Reply:
Not sure where you are looking, but this is what you are trying to merge ("files changed" tab)

![image](https://user-images.githubusercontent.com/2486142/197348514-70d101c9-a8f0-4416-b4b8-e3c996f557df.png)

Reply:
Now I'm confused lol I thought we wanted to remove provider from dependency array?

Reply:
Not from the dep array. You were calling `fetchPriceFromUniswap` with `provider` as an argument. And we don't need it to pass it over, since it's already in the state.

I think the code I pasted above should do it.

Reply:
Gotcha 😝